src/main/java/edu/kit/provideq/toolbox/format/cnf/dimacs/DimacsCnf.java

```diff
@@ -38,8 +38,15 @@ public DimacsCnf(ArrayList<ArrayList<Variable>> orClauses) {
   }
 
   public DimacsCnf(ArrayList<ArrayList<Variable>> orClauses, List<Variable> variables) {
-    this.variables = List.copyOf(variables);
-    this.orClauses = new ArrayList<>(orClauses);
+    this.variables = variables.stream()
+            .map(Variable::new)
+            .collect(Collectors.toCollection(ArrayList::new));
+
+    this.orClauses = orClauses.stream()
+            .map(list -> list.stream()
+                    .map(Variable::new)
+                    .collect(Collectors.toCollection(ArrayList::new)))
```

Comment:
please apply the DRY principle here and extract the List<Variable> copying to its own (private) function